app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/WorkoutAnalytics.java

```diff
@@ -0,0 +1,423 @@
+        //DURATION
+        //
+
+        String time_formatted = df.format(d);
+        setTimeDurationText.setText(time_formatted);
+
+        //
+        //AVERAGE WORKOUT TIME
+        //
+
+        float Avg_Workout_Time= ((float)difference / Float.valueOf(1))/MILLISECONDS_IN_SECOND;
+        String avg_time_str = String.format("%.2f sec",(Avg_Workout_Time/workoutsBetweenDateRange.size()));
+        setAvgWorkoutTimeText.setText(avg_time_str);
+
+    }
+
+
+
+
+
+
+    public void createBarGraph(String Date1, String Date2){
+        int count;
+        int totaldistance;
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MM/dd/yyyy");
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        try {
+            Date date1 = simpleDateFormat.parse(Date1);
+            Date date2 = simpleDateFormat.parse(Date2);
+
+            Calendar mDate1 = Calendar.getInstance();
+            Calendar mDate2 = Calendar.getInstance();
+            mDate1.clear();
+            mDate2.clear();
+
+            mDate1.setTime(date1);
+            mDate2.setTime(date2);
+
+            dates = new ArrayList<>();
+            dates = getList(mDate1,mDate2);
+
+
+            String tempStartDate;
+            barEntries = new ArrayList<>();
+
+
+
+            for(int j = 0; j< dates.size();j++){
+                count = 0;
+                totaldistance = 0;
+//                Log.d(TAG, "Test" + dates.get(j));
+
+                for(int i = 0; i<=workoutsBetweenDateRange.size()-1;i++) {
+
+                    tempStartDate = simpleDateFormat.format(workoutsBetweenDateRange.get(i).getStartDate());
+
+
+                    if (dates.get(j).equals(tempStartDate)) {
+
+                        if((unit_abbrev.compareTo(" m") !=0 )&&(workoutsBetweenDateRange.get(i).getPoolUnits().compareTo("Meters") == 0)){
+                            totaldistance += (workoutsBetweenDateRange.get(i).getTotalDistanceTraveled()*METERTOYARDCONVERT);
+                        }
+                        else if((unit_abbrev.compareTo(" yd") !=0 )&&(workoutsBetweenDateRange.get(i).getPoolUnits().compareTo("Yards") == 0)){
+                            totaldistance += (workoutsBetweenDateRange.get(i).getTotalDistanceTraveled()*YARDTOMETERCONVERT);
+                        }
+                        else {
+                            totaldistance += workoutsBetweenDateRange.get(i).getTotalDistanceTraveled();
+                        }
+
+                        count++;
+                    }
+                }
+                if (count == 0) {
+                    count = 1;
+                }
+
+                totaldistance = totaldistance / count;
+                barEntries.add(new BarEntry(totaldistance,j));
+            }
+
+        }catch(ParseException e){
+            e.printStackTrace();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        BarDataSet barDataSet = new BarDataSet(barEntries,"Distance Swam");
```

Comment:
Rename this to "Distance Traveled", the typical English confusion of swam/swum makes this sound weird even though I think it's the correct usage?

Also, please use a string resource.